### ISSUE TITLE
Keybase requires reprovisioning after update?

### ISSUE THREAD

**Issue by ShaneMcC** [keybase/client#3850]
Hi,

I used keybase a while ago on my machine (around December last year) got it all set up and working - great.

Went to play with it again today and it says I need to login on the device again, but logging in complains that I can't reuse the same Device name.

Looking in the ~/.config/keybase dir, I have:

> -rw------- 1 shane shane  290 Dec 16  2015 config.json.########################
> -rw------- 1 shane shane 1380 Dec 16  2015 secretkeys.dataforce.mpack

Where the ######################## is a random-looking ID so I've masked it here for now. Of note is that there doesn't appear to be a config.json file without the random extension.

Looking in that file, there is a user section with a "device" and a device id - though this doesn't appear to match the device ID used in the "device remove" example at https://keybase.io/dataforce/devices ?

Given that this device hasn't been reinstalled and the .config/keybase directory hasn't changed i'm curious as to what has happened here and why this device no longer seems to think it is linked to my account, and also how I can go about resolving this (Trying to re-add it complains that the device name is already in use, which is True, but it's the same device)

Thanks,

- Shane

**Comment by cjb** [comment]
Thanks for the report.  Please could you run `keybase log send` and tell us the ID?

CC @patrickxb @maxtaco

**Comment by ShaneMcC** [comment]
Hi,

The log id: d21f1facf5cdb4a1e331811c

Thanks,

- Shane

**Comment by cjb** [comment]
Thanks.  How did you install it?  I guess you haven't updated it since December, is that right?  We'll check it out.

**Comment by ShaneMcC** [comment]
Hi,

I can't remember how I originally installed it, presumably through apt package manager given that it seems to have kept itself up to date that way:

> [15:44:28] [shane@Xion:~]$ dpkg -l | grep -i keybase
> ii  keybase                      1.0.16                amd64        Keybase's command line client
> [15:44:29] [shane@Xion:~]$

I've not touched keybase since originally adding the device in December.

Thanks,

- Shane

**Comment by cjb** [comment]
Oops.  Versions like:

ii  keybase        1.0.16-20160 amd64        Keybase's command line client

**Comment by maxtaco** [comment]
Thanks @ShaneMcC.  Seems like we couldn't find your config file, which is here, and hasn't changed in a long time:

/home/shane/.config/keybase/config.json

Do you have a file there?  What does it say?

**Comment by cjb** [comment]
@maxtaco See earlier in the thread -- it looks like it was renamed to config.json.<random string> somehow.

**Comment by patrickxb** [comment]
I commented in JIRA.

line 2 of log, can't find config.json.  Without config.json, need to provision.

during provisioning a new config.json.XXXXX is created but never swapped in because provisioning doesn't complete.  This is by design.

**Comment by ShaneMcC** [comment]
@cjb:

> [15:44:29] [shane@Xion:~]$ apt-cache showpkg keybase
> Package: keybase
> Versions:
> 1.0.16 (/var/lib/apt/lists/dist.keybase.io_linux_deb_repo_dists_stable_main_binary-amd64_Packages) (/var/lib/dpkg/status)
>  Description Language:
>                  File: /var/lib/apt/lists/dist.keybase.io_linux_deb_repo_dists_stable_main_binary-amd64_Packages
>                   MD5: f5b8aebb13380e4248f4880bb5d8d942
>  Description Language:
>                  File: /var/lib/apt/lists/dist.keybase.io_linux_deb_repo_dists_stable_main_binary-i386_Packages
>                   MD5: f5b8aebb13380e4248f4880bb5d8d942
> 
> 
> Reverse Depends:
>   keybase:i386,keybase
> Dependencies:
> 1.0.16 - keybase:i386 (32 (null))
> Provides:
> 1.0.16 -
> Reverse Provides:
> [15:46:45] [shane@Xion:~]$ cat /var/lib/apt/lists/dist.keybase.io_linux_deb_repo_dists_stable_main_binary-amd64_Packages
> Package: keybase
> Version: 1.0.16
> Architecture: amd64
> Installed-Size: 36568
> Maintainer: Keybase Bugs <bugs@keybase.io>
> Priority: optional
> Section: misc
> Filename: pool/main/k/keybase/keybase_1.0.16_amd64.deb
> Size: 6127178
> SHA256: f17c08051715726b6ca95d11278d987b4c0732af02e4477dd2ef41236db625f1
> SHA1: b0b30d32507c9ef89aac6a681651c933b577051d
> MD5sum: 7c7eef3287979f6856d5c8a8bd7cd36a
> Description: Keybase's command line client
> 
> [15:46:58] [shane@Xion:~]$ cat /etc/apt/sources.list.d/keybase.list
> ### THIS FILE IS AUTOMATICALLY CONFIGURED ###
> # You may comment out this entry, but any other modifications may be lost.
> deb http://dist.keybase.io/linux/deb/repo stable main
> [15:47:22] [shane@Xion:~]$ ls -al /etc/apt/sources.list.d/keybase.list
> -rw-r--r-- 1 root root 175 Jul 31 19:40 /etc/apt/sources.list.d/keybase.list
> [15:47:27] [shane@Xion:~]$

@maxtaco:

No such file there, just the oddly named one, the directory listing at the top is the entire contents of /home/shane/.config/keybase/

I've tried renaming the file to just config.json but this doesn't appear to help (it only changes `keybase status` to show the correct username), it still shows me as logged out and requires me to login and the same occurs wrt to needing to re-add the device.

**Comment by cjb** [comment]
@patrickxb I wonder in what sense it didn't complete, given that there's a secretkeys file and the API server has registered the device.

**Comment by maxtaco** [comment]
Sorry I wasn't paying enough attention, we're discussing it....

**Comment by patrickxb** [comment]
@cjb No, the current provisioning that it started today.  Not the one from december.  That one worked fine.  max and I discussing.  no need to pollute this thread.

**Comment by cjb** [comment]
@patrickxb Given:
```
-rw------- 1 shane shane 290 Dec 16 2015 config.json.########################
```
The config.json with the random ending is not from today, it is from Dec 16 2015.

I'll bow out of the thread now, thanks.

**Comment by patrickxb** [comment]
Sorry, I was looking at the log, which shows a new config.json.XXXX being created.

**Comment by maxtaco** [comment]
We're stumped, it seems like something deleted your old `config.json`. You can try this:

```sh
keybase ctl stop
cat << EOF > ~/.config/keybsae/config.json
{
    "current_user": "dataforce",
    "users": {
        "max": {
            "device": "15a9fcf4e247568bdf4168e5f09ac918",
            "id": "f732170fa769a0c34bf248317f478e00",
            "name": "dataforce",
            "salt": "1a507c4a3c083e6a6e72d79f71ea99e8"
        }
    }
}
keybase ctl start
keybase login
```

This is all public info based on your username/UID.

**Comment by patrickxb** [comment]
replace "max" with "dataforce"

**Comment by ShaneMcC** [comment]
That's done the trick (well, once I changed "max" to "dataforce" and used keybase rather than keybsae ;) ), thanks - that keeps my naming-OCD happy :)

Out of interest, where did you get the salt from? (I can see the other bits from https://keybase.io/dataforce/sigchain#4361ff1069fa3b4102adf5cdf74e0cd434e257d0f1d35bb60a580f650a60d9c90f)

**Comment by maxtaco** [comment]
Urg, got it right in ~3 places, missed the 4th! 

All users' salts are public:

```
https://keybase.io/_/api/1.0/getsalt.json?username=dataforce
```

Since it's the first step in the password stretching process, and it happens before you've authenticated.
